back-end/src/sciler/config/checkConfig.go

```diff
@@ -0,0 +1,367 @@
+			return constraint.checkConstraintsDeviceType(inputType, ruleID)
+		} else if outputObject, ok := device.Output[constraint.ComponentID]; ok {
+			return constraint.checkConstraintsDeviceType(outputObject.Type, ruleID)
+		} else {
+			return []string{fmt.Sprintf("on rule %s: component with id %s not found in map", ruleID, constraint.ComponentID)}
+		}
+	} else {
+		return []string{fmt.Sprintf("on rule %s: device with id %s not found in map", ruleID, condition.TypeID)}
+	}
+}
+
+// checkConstraintsTimer is a method that check all types and comparators of a constraint on a timer
+func checkConstraintsTimer(condition Condition, config WorkingConfig, ruleID string, constraint Constraint) []string {
+	if _, ok := config.Timers[condition.TypeID]; ok {
+		return checkConstraintsBooleanInput(ruleID, constraint)
+	}
+	return []string{fmt.Sprintf("on rule %s: timer with id %s not found in map", ruleID, condition.TypeID)}
+}
+
+// checkConstrainsRule is a method that check all types and comparators of a constraint on a rule
+func checkConstrainsRule(condition Condition, config WorkingConfig, ruleID string, constraint Constraint) []string {
+	if _, ok := config.RuleMap[condition.TypeID]; ok { // checks if rule can be found in the map, if so, it is stored in variable device
+		valueType := reflect.TypeOf(constraint.Value).Kind()
+		comparison := constraint.Comparison
+		if valueType != reflect.Int && valueType != reflect.Float64 {
+			return []string{fmt.Sprintf("on rule %s: value type numeric expected but %s found as type of value %v", ruleID, valueType.String(), constraint.Value)}
+		}
+		if !checkValidComparison(comparison) {
+			return []string{fmt.Sprintf("on rule %s: comparison %s is not valid", ruleID, comparison)}
+		}
+		if comparison == "contains" {
+			return []string{fmt.Sprintf("on rule %s: comparison %s not allowed on rule", ruleID, comparison)}
+		}
+	} else {
+		return []string{fmt.Sprintf("on rule %s: rule with id %s not found in map", ruleID, condition.TypeID)}
+	}
+	// all cases for errors are already handled
+	return make([]string, 0)
+}
+
+// checkConstraintsDeviceType is a method that check the input type and check if the constraint has that value type
+func (constraint Constraint) checkConstraintsDeviceType(inputType string, ruleID string) []string {
+	switch inputType {
+	case "string":
+		return checkConstraintsDeviceStringInput(ruleID, constraint)
+	case "boolean":
+		return checkConstraintsBooleanInput(ruleID, constraint)
+	case "numeric":
+		return checkConstraintsDeviceNumericInput(ruleID, constraint)
+	case "array":
+		return checkConstraintsDeviceArrayInput(ruleID, constraint)
+	default:
+		return []string{fmt.Sprintf("on rule %s: custom types like: %s, are not yet implemented", ruleID, inputType)}
+	}
+}
+
+// checkConstraintsDeviceStringInput is a method that returns all error (if any) in a constraint of a device with string input
+func checkConstraintsDeviceStringInput(ruleID string, constraint Constraint) []string {
```

Comment:
these are not only for input so `checkConstraintsDeviceStringType` -> also for all methods below

Reply:
also new line please

Reply:
fixed